drivers/dma/dma_silabs_siwx917.c

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/dma.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/types.h>
+#include "rsi_rom_udma_wrapper.h"
+#include "rsi_udma.h"
+#include "sl_status.h"
+
+#define DT_DRV_COMPAT          silabs_siwx917_dma
+#define DMA_MAX_TRANSFER_COUNT 1024
+#define DMA_CH_PRIORITY_HIGH   1
+#define DMA_CH_PRIORITY_LOW    0
+#define VALID_BURST_LENGTH     0
+#define UDMA_ADDR_INC_NONE     0X03
+
+LOG_MODULE_REGISTER(si91x_dma, CONFIG_DMA_LOG_LEVEL);
+
+struct dma_siwx917_config {
+	UDMA0_Type *reg;                 /* UDMA register base address */
+	uint8_t channels;                /* UDMA channel count */
+	uint8_t irq_number;              /* IRQ number */
+	RSI_UDMA_DESC_T *sram_desc_addr; /* SRAM Address for UDMA Descriptor Storage */
+	void (*irq_configure)(void);     /* IRQ configure function */
+};
+
+struct dma_siwx917_data {
+	UDMA_Channel_Info *chan_info;
+	dma_callback_t dma_callback;            /* User callback */
+	void *cb_data;                          /* User callback data */
+	RSI_UDMA_DATACONTEXT_T dma_rom_buff[1]; /* Buffer to store UDMA handle */
```

Comment:
What is the purpose of declaring an array on 1 item? You will be able to get a reference on this member even if it is not an array:

    void *xxx = &data->dma_rom_buff;

Reply:
Removed array declaration.